src/event-bridge/utils.ts

```diff
@@ -184,7 +185,7 @@ export const flattenExpression = (expr: Expr, scope: EventScope): Expr => {
       throw Error("Object access must be a string.");
     } else if (isArrayLiteralExpr(parent)) {
       if (typeof key === "number") {
-        return parent.items[key];
+        return parent.items[key] ?? new UndefinedLiteralExpr();
```

Comment:
should use `parent.fork(new UndefinedLiteralExpr())` or else the undefined literal will be an orphaned node.